Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -664,212 +668,108 @@ class InAppMessageViewTest: IntegrationTest {
-        XCTAssertTrue(eventListenerMock.messageShownCalled)
-        XCTAssertFalse(eventListenerMock.messageDismissedCalled)
-        XCTAssertFalse(eventListenerMock.errorWithMessageCalled)
-
-        // Tap button again
-        onCustomActionButtonPressed(onInlineView: inlineView)
-        XCTAssertEqual(eventListenerMock.messageActionTakenCallsCount, 2)
-    }
+    // MARK: global event listener
 
     @MainActor
-    func test_onInlineMessageShown_givenDelegateSet_onMultipleButtonTap_expectSingleTrackClickedMetric() async {
-        DIGraphShared.shared.override(value: eventBusHandlerMock, forType: EventBusHandler.self)
-
-        let inlineView = await showInlineMessageForMetrics(setDelegate: true)
-        XCTAssertTrue(isInlineViewVisible(inlineView))
-        onCustomActionButtonPressed(onInlineView: inlineView)
+    func test_onInlineMessageShown_expectGlobalMessageShownListener() async {
+        let givenInlineMessage = Message.randomInline
+        queueMock.getInlineMessagesReturnValue = [givenInlineMessage]
 
-        XCTAssertEqual(eventBusHandlerMock.postEventCallsCount, 2)
+        let inlineView = InAppMessageView(elementId: givenInlineMessage.elementId!)
+        await onDoneRenderingInAppMessage(givenInlineMessage, insideOfInlineView: inlineView)
 
-        // Tap the button again and verify that
-        // the call to onActionClick updates
-        onCustomActionButtonPressed(onInlineView: inlineView)
-        XCTAssertEqual(eventBusHandlerMock.postEventCallsCount, 3)
+        assert(message: givenInlineMessage, didCallMessageShownEventListener: true)
     }
 
+    // Tests a scenario where multiple inline messages are displayed to the user,
+    // then expect multiple MessageShown global event listeners to be called for each message shown.
     @MainActor
-    func test_onInlineMessageShown_givenDelegateSet_onMultipleButtonTap_expectGlobalMessageActionTakenCallback() async {
-        DIGraphShared.shared.override(value: eventBusHandlerMock, forType: EventBusHandler.self)
-
-        let inlineView = await showInlineMessageForMetrics(setDelegate: true)
-        XCTAssertTrue(isInlineViewVisible(inlineView))
-        onCustomActionButtonPressed(onInlineView: inlineView)
-
-        XCTAssertTrue(inlineMessageDelegateMock.onActionClickCalled)
-        XCTAssertEqual(inlineMessageDelegateMock.onActionClickCallsCount, 1)
+    func test_givenMultipleInlineMessageRendered_expectTrackMultipleMessageShownListenerCalls() async {
+        // FIXME: This test is failing because the queueMock is being mocked and is not a real instance.
+        // the queue's real instance is able to return a message by elemenetid. when you mock the queue, it will
+        // ignore the elementid and always return the same message.
+        // So, this test wants each inline View to have a separate message but instead each view will get the same message.
+        // PR that would fix this test: https://github.com/customerio/customerio-ios/pull/776
```

Comment:
I think this is a good test function. Unfortunately, because the test class mocks the queue, the test will always fail.

If the PR 776 gets merged first, I am OK with merging that code into this PR and removing this `FIXME` comment before merging. Or, if needed, we can comment out this test and uncomment it after that 776 gets merged.

Reply:
For now, I will comment out this test method and will uncomment after PR#776 is merged.